src/awkward/_v2/contents/numpyarray.py

```diff
@@ -623,6 +626,8 @@ def _unique(self, negaxis, starts, parents, outlength):
         if negaxis is None:
             contiguous_self = self if self.is_contiguous else self.contiguous()
             flattened_shape = 1
+            # TODO: could this be contiguous_self.shape[i]?
+            # pylint: disable-next=consider-using-enumerate
             for i in range(len(contiguous_self.shape)):
                 flattened_shape = flattened_shape * self.shape[i]
```

Comment:
It could. `self` and `contiguous_self` are guaranteed to have the same `shape`.

```suggestion
            flattened_shape = 1
            for x in enumerate(contiguous_self.shape):
                flattened_shape = flattened_shape * x
```

Reply:
This could be `math.prod(contiguous_self.shape)` in Python 3.8+. Is there a reason this isn't `np.prod(contiguous_self.shape)`? (small array in a hot loop?)

Reply:
`np.prod` could do it, but in this context, we want to avoid confusion between this small-scale tuple arithmetic and arrays that are dealt with via nplike. This is one that would have to happen through actual NumPy, not CuPy or whatever, but then it would stand out as a possible mistake of using NumPy instead of nplike.

There are other, similar, situations that aren't as simple as a straight product, too. There's some value for this one to look like the others.

Reply:
It being a (small) tuple vs. an array would be enough for me to avoid `np.prod`. Though `math.prod` would be nice, eventually. There's value in simple things looking and reading simply, with special cases standing out. For example, when I read it, I wasn't sure if it could be reduced to a simple product.